### DISCUSSION TITLE
Recommended way to dynamically apply overlays for custom Tegra SoM carrier boards

### DISCUSSION THREAD

**Discussion by owestermann** [OE4T/meta-tegra#1509] [Q&A]
Hey,

we're looking into a custom carrier board for the NVidia Tegra SoM modules, specifically the Orin Nano. We've a yocto setup which builds our software and we can flash it on both the NVidia Jetson Orin Nano EVK board as well as onto our own board.

We would like to support some variants of our custom hardware, which would need different device tree overlays to configure e.g. I2C devices etc. We've stumble across the NVidia mechanism which does seemingly check the Jetson Orin Nano carrier board for an EEPROM and it's content, but currently we're lacking the bigger picture and how for example the camera overlays are applied.

Is there some existing mechanism in UEFI/L4T/... which we could extend to add our own hardware detection and selection of DTB? If not, is there some recommended way to implement this?

Disclaimer: I'm coming from a world of U-Boot and self-build fitconfig selection in there, so I might be missing something very obvious ;-)

**Comment by madisongh** [comment]
The EEPROM-based mechanism that NVIDIA provides is the only one I know of that they build into their bootloaders.  The [Jetson Linux Developer's Guide](https://docs.nvidia.com/jetson/archives/r35.5.0/DeveloperGuide/index.html) describes the EEPROM layout and how the plugin manager applies the overlays - to some degree, at least.  

If you're feeling brave, you could, I guess, customize UEFI to implement your own mechanism, too.
